attribute-masking-service/README.md

```diff
@@ -16,13 +16,17 @@ limitations under the License.
 
 # <img src="../logos/logo.svg" width="180">
 
-# Palisade Attribute-Masking-Service
+# Flow of Control
 
-The attribute-masking-service is the final transformation the palisade system applies to resources before they are returned.
-The service performs two functions:
-* Store the full details of the authorised request in a persistence store, to be later retrieved by the data-service
-* Mask the leafResource, removing any sensitive information
+![Attribute-Masking Service Diagram](doc/attribute-masking-service.png)
 
+The Attribute-Masking Service will persist authorised requests and forward a redacted version of the resource onto the next service on the pipeline.
+This is an interim service on the Palisade stream pipeline that reads each resource sent from the Policy Service. 
+It will take each of these resources, ascertain if it is a "start" or "end" marker for the set of records (has a token, but the message body is empty of content).
+If it is, no further processing is done.
+For messages that represent records, these are persisted as authorised requests and then a masked version of the resource is prepared to be forwarded to the next services.
+Both types of messages (markers and records) are forwarded to both the Topic-Offset Service and the Filtered-Resource Service.
+If at any time during the processing of a message, an error occurs, an error message is produced and sent to the Audit Service.
```

Comment:
This doesn't read right as the word "resource" appears in different contexts.

```suggestion
The Attribute-Masking Service will persist authorised requests which will be checked by the Data Service.
This service reads messages from the `rule` Kafka topic. 
It the message has a "start" or "end" marker for the set of records (has a token, but the message body is empty of content), no further processing is done and these are added to the `masked-resource` Kafka topic.
For messages that represent a record, these are persisted as authorised requests and any resource level rules are applied to remove a resource from the message stream that cannot be accessed by the user.
The `masked-resource` Kafka topic is read by both the Filtered-Resource Service and the Topic-Offset Service.
If at any time during the processing of a message, an error occurs, an AuditErrorMessage is produced and added to the `error` topic to be consumed by the Audit Service.
```

Reply:
Have taken some of the ideas and changed it around a bit.